src/core/qgslegendrenderer.cpp

```diff
@@ -630,15 +630,6 @@ QgsLegendRenderer::LegendComponent QgsLegendRenderer::drawSymbolItemInternal( Qg
 {
   QgsLayerTreeModelLegendNode::ItemContext ctx;
   ctx.context = context;
-
-  // add a layer expression context scope
-  QgsExpressionContextScope *layerScope = nullptr;
```

Comment:
Are these changes here done on purpose?
Be aware that this class may also be used in server or 3rd party applications, so it's not possible to shuffle around context generation to somewhere else in the call stack.

Reply:
Yes. Originally the PR done by Nyall laid the foundation for this PR. This part that Mr. Dawson added is serving the same purpose as what I have doing but in a more limited way. 

I felt that letting this in would end up evaluating the output of the evaluation twice. To prevent this double evaluation this was removed.

Reply:
> the PR done by Nyall

Link

> I felt that letting this in would end up evaluating the output of the evaluation twice. To prevent this double evaluation this was removed.

I don't see any evaluation here. Just a scope that is no longer added in here. This carries a risk that a codepath (thinking e.g. of server) leading here will lack the scope in the future. Convince me that this risk is void and that the new approach is "better".

Reply:
you're right, I'll readd this in case there are some other uses to this.